src/test/java/de/tum/cit/aet/artemis/exercise/programming/AuxiliaryRepositoryResourceIntegrationTest.java

```diff
@@ -0,0 +1,519 @@
+package de.tum.cit.aet.artemis.exercise.programming;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.argThat;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.commons.io.FileUtils;
+import org.eclipse.jgit.api.ListBranchCommand;
+import org.eclipse.jgit.api.MergeResult;
+import org.eclipse.jgit.lib.Ref;
+import org.eclipse.jgit.merge.MergeStrategy;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.util.LinkedMultiValueMap;
+
+import de.tum.cit.aet.artemis.AbstractSpringIntegrationJenkinsGitlabTest;
+import de.tum.cit.aet.artemis.core.domain.Course;
+import de.tum.cit.aet.artemis.programming.domain.AuxiliaryRepository;
+import de.tum.cit.aet.artemis.programming.domain.File;
+import de.tum.cit.aet.artemis.programming.domain.FileType;
+import de.tum.cit.aet.artemis.programming.domain.ProgrammingExercise;
+import de.tum.cit.aet.artemis.programming.domain.Repository;
+import de.tum.cit.aet.artemis.programming.dto.FileMove;
+import de.tum.cit.aet.artemis.programming.dto.RepositoryStatusDTO;
+import de.tum.cit.aet.artemis.programming.repository.AuxiliaryRepositoryRepository;
+import de.tum.cit.aet.artemis.programming.repository.ProgrammingExerciseBuildConfigRepository;
+import de.tum.cit.aet.artemis.programming.repository.ProgrammingExerciseRepository;
+import de.tum.cit.aet.artemis.programming.service.GitService;
+import de.tum.cit.aet.artemis.programming.web.repository.FileSubmission;
+import de.tum.cit.aet.artemis.util.GitUtilService;
+import de.tum.cit.aet.artemis.util.LocalRepository;
+
+class AuxiliaryRepositoryResourceIntegrationTest extends AbstractSpringIntegrationJenkinsGitlabTest {
```

Comment:
I think #9314 broke the imports here. Can you have a look?

```
> Task :compileTestJava
/home/runner/work/Artemis/Artemis/src/test/java/de/tum/cit/aet/artemis/exercise/programming/AuxiliaryRepositoryResourceIntegrationTest.java:34: error: cannot find symbol
import de.tum.cit.aet.artemis.AbstractSpringIntegrationJenkinsGitlabTest;
                             ^
  symbol:   class AbstractSpringIntegrationJenkinsGitlabTest
  location: package de.tum.cit.aet.artemis
/home/runner/work/Artemis/Artemis/src/test/java/de/tum/cit/aet/artemis/exercise/programming/AuxiliaryRepositoryResourceIntegrationTest.java:48: error: package de.tum.cit.aet.artemis.util does not exist
import de.tum.cit.aet.artemis.util.GitUtilService;
                                  ^
/home/runner/work/Artemis/Artemis/src/test/java/de/tum/cit/aet/artemis/exercise/programming/AuxiliaryRepositoryResourceIntegrationTest.java:49: error: package de.tum.cit.aet.artemis.util does not exist
import de.tum.cit.aet.artemis.util.LocalRepository;
                                  ^
/home/runner/work/Artemis/Artemis/src/test/java/de/tum/cit/aet/artemis/exercise/programming/AuxiliaryRepositoryResourceIntegrationTest.java:51: error: cannot find symbol
class AuxiliaryRepositoryResourceIntegrationTest extends AbstractSpringIntegrationJenkinsGitlabTest {
                                                         ^
  symbol: class AbstractSpringIntegrationJenkinsGitlabTest
/home/runner/work/Artemis/Artemis/src/test/java/de/tum/cit/aet/artemis/exercise/programming/AuxiliaryRepositoryResourceIntegrationTest.java:76: error: cannot find symbol
    private final LocalRepository localAuxiliaryRepo = new LocalRepository(defaultBranch);
```

Reply:
Yep, the imports were indeed broken. I fixed them, now the tests work again :+1: